### PR TITLE
fix modal submit sending data while button says skip

### PR DESCRIPTION
**Pull request by npretto**
I probably got lost in all the changes, but it should not have sent the feedback if empty

See slack thread: https://metaboat.slack.com/archives/C063Q3F1HPF/p1712684011556149?thread_ts=1712319863.930869&cid=C063Q3F1HPF

### CODE REVIEW

File: frontend/src/metabase/home/components/EmbedHomepage/tests/common.unit.spec.tsx [hunk 1]
```diff
@@ -169,14 +169,10 @@ describe("EmbedHomepage (OSS)", () => {
...
 
-      const feedbackBody = await getLastFeedbackCall()?.request?.json();
-
-      expect(feedbackBody).toEqual({
-        source: "embedding-homepage-dismiss",
-      });
-
       await waitForElementToBeRemoved(() => queryFeedbackModal());
 
+      expect(getLastFeedbackCall()).toBeUndefined();
```
Comment (npretto) [inline frontend/src/metabase/home/components/EmbedHomepage/tests/common.unit.spec.tsx:174]: I moved it after the waitFor to make sure the request is not just delayed.
I checked and this test fails without the fix

Comment (heypoom) [reply frontend/src/metabase/home/components/EmbedHomepage/tests/common.unit.spec.tsx:174]: I think adding in reasoning like this to the code comment itself is quite useful for later editors, as it's not immediately clear to me what `expect(getLastFeedbackCall()).toBeUndefined()` is for, or supposed to do

**Review by heypoom — APPROVED** [review]
Looks good! 👍🏻

File: frontend/src/metabase/home/components/EmbedHomepage/tests/common.unit.spec.tsx (hunk 1 above)
Comment (npretto) [reply frontend/src/metabase/home/components/EmbedHomepage/tests/common.unit.spec.tsx:176]: Thanks, added an explanation

**Pull request merged**